server/service/integration_live_queries_test.go

```diff
@@ -153,15 +180,57 @@ func (s *liveQueriesTestSuite) TestLiveQueriesRestOneHostOneQuery() {
 
 		wg.Wait()
 
-		require.Len(t, liveQueryResp.Results, 1)
-		assert.Equal(t, 1, liveQueryResp.Summary.RespondedHostCount)
-		assert.Equal(t, q1.ID, liveQueryResp.Results[0].QueryID)
-		require.Len(t, liveQueryResp.Results[0].Results[0].Rows, 1)
-		assert.Equal(t, "a", liveQueryResp.Results[0].Results[0].Rows[0]["col1"])
-		assert.Equal(t, "b", liveQueryResp.Results[0].Results[0].Rows[0]["col2"])
+		var result fleet.QueryResult
+		if newEndpoint {
+			assert.Equal(t, q1.ID, oneLiveQueryResp.QueryID)
+			assert.Equal(t, 1, oneLiveQueryResp.TargetedHostCount)
+			assert.Equal(t, 1, oneLiveQueryResp.RespondedHostCount)
+			require.Len(t, oneLiveQueryResp.Results, 1)
+			result = oneLiveQueryResp.Results[0]
+		} else {
+			require.Len(t, liveQueryResp.Results, 1)
+			assert.Equal(t, 1, liveQueryResp.Summary.TargetedHostCount)
+			assert.Equal(t, 1, liveQueryResp.Summary.RespondedHostCount)
+			assert.Equal(t, q1.ID, liveQueryResp.Results[0].QueryID)
+			require.Len(t, liveQueryResp.Results[0].Results, 1)
+			result = liveQueryResp.Results[0].Results[0]
+		}
+		assert.Equal(t, host.ID, result.HostID)
+		require.Len(t, result.Rows, 1)
+		assert.Equal(t, "a", result.Rows[0]["col1"])
+		assert.Equal(t, "b", result.Rows[0]["col2"])
+
+		// For loop, waiting for activity feed to update, which happens after aggregated stats update.
+		var activity *fleet.ActivityTypeLiveQuery
+		activityUpdated := make(chan *fleet.ActivityTypeLiveQuery)
+		go func() {
+			for {
+				details := json.RawMessage{}
+				err := mysql.ExecAdhocSQLWithError(
+					s.ds, func(q sqlx.ExtContext) error {
+						return sqlx.GetContext(
+							context.Background(), q, &details,
+							`SELECT details FROM activities WHERE activity_type = 'live_query' ORDER BY id DESC LIMIT 1`,
+						)
+					},
+				)
+				if !errors.Is(err, sql.ErrNoRows) {
+					act := fleet.ActivityTypeLiveQuery{}
+					err = json.Unmarshal(details, &act)
+					require.NoError(t, err)
+					if act.QuerySQL == q1.Query {
+						assert.Equal(t, act.TargetsCount, uint(1))
+						activityUpdated <- &act
+					}
+				}
```

Comment:
Nit: There's `lastActivityOfTypeMatches` (maybe it's useful here?)

Reply:
I like the approach of `lastActivityOfTypeMatches`, but it needs an exact details match and ends the test if it doesn't find one.

A related thing I'm concerned about here is if tests run in parallel and some other test writes a new activity into the test DB, which will cause this test to fail. Is that possible?